sapi/fpm/fpm/fpm_stdio.c

```diff
@@ -162,49 +162,52 @@ static void fpm_stdio_child_said(struct fpm_event_s *ev, short which, void *arg)
 		}
 	}
 
-	while (fifo_in || fifo_out) {
-		if (fifo_in) {
-			res = read(fd, buf + in_buf, max_buf_size - 1 - in_buf);
-			if (res <= 0) { /* no data */
-				fifo_in = 0;
-				if (res == 0 || (errno != EAGAIN && errno != EWOULDBLOCK)) {
-					/* pipe is closed or error */
-					read_fail = (res < 0) ? res : 1;
-				}
-			} else {
-				in_buf += res;
-				/* if buffer ends with \0, then the stream will be finished */
-				if (!buf[in_buf - 1]) {
-					finish_log_stream = 1;
-					in_buf--;
-				}
+	while (1) {
+stdio_read:
+		in_buf = read(fd, buf, max_buf_size - 1);
+		if (in_buf <= 0) { /* no data */
+			if (in_buf == 0 || (errno != EAGAIN && errno != EWOULDBLOCK)) {
+				/* pipe is closed or error */
+				read_fail = (in_buf < 0) ? in_buf : 1;
 			}
+			break;
 		}
-
-		if (fifo_out) {
-			if (in_buf == 0) {
-				fifo_out = 0;
+		start = 0;
+		if (cmd_pos > 0) {
+			if 	(!memcmp(buf, &FPM_STDIO_CMD_FLUSH[cmd_pos], sizeof(FPM_STDIO_CMD_FLUSH) - cmd_pos)) {
```

Comment:
What if there are less than `sizeof(FPM_STDIO_CMD_FLUSH) - cmd_pos)` characters in buf?

Reply:
Good catch! Fixed!